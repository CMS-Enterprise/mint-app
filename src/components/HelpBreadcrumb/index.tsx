import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Button,
  IconClose
} from '@trussworks/react-uswds';
import classNames from 'classnames';

import UswdsReactLink from 'components/LinkWrapper';

type HelpBreadcrumbProps = {
  className?: string;
  home?: boolean;
  text?: string;
  newTabOnly?: boolean;
};

export default function HelpBreadcrumb({
  className,
  home,
  text,
  newTabOnly = false
}: HelpBreadcrumbProps) {
  const history = useHistory();
  const newTab = history.length === 1;
  const { t } = useTranslation('helpAndKnowledge');
  const handleClick = () => {
    if (newTab || newTabOnly) {
      window.close();
    } else {
      history.push(home ? '/' : '/help-and-knowledge');
    }
  };
  return (
    <Button
      type="button"
      unstyled
      onClick={() => handleClick()}
      className={classNames(className)}
    >
      {newTab || newTabOnly ? (
        <>
          <IconClose className="margin-right-05 margin-top-6 text-tbottom" />
          {t('close')}
        </>
      ) : (
        <BreadcrumbBar variant="wrap">
          <Breadcrumb>
            <BreadcrumbLink asCustom={UswdsReactLink} to="/">
              <span>{home ? t('home') : t('heading')}</span>
            </BreadcrumbLink>
          </Breadcrumb>
          <Breadcrumb current>{text}</Breadcrumb>
        </BreadcrumbBar>
      )}
    </Button>
  );
}
