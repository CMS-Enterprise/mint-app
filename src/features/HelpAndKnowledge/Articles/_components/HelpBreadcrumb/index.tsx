import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { Button, Icon } from '@trussworks/react-uswds';
import classNames from 'classnames';

import Breadcrumbs, { BreadcrumbItemOptions } from 'components/Breadcrumbs';

type HelpBreadcrumbProps = {
  className?: string;
  home?: boolean;
  text?: string;
  newTabOnly?: boolean;
};

export default function HelpBreadcrumb({
  className,
  home,
  text,
  newTabOnly = false
}: HelpBreadcrumbProps) {
  const history = useHistory();
  const newTab = history.length === 1;
  const { t } = useTranslation('helpAndKnowledge');
  const handleClick = () => {
    if (newTab || newTabOnly) {
      window.close();
    } else {
      history.replace(home ? '/' : '/help-and-knowledge');
    }
  };

  return (
    <Button
      type="button"
      unstyled
      onClick={() => handleClick()}
      className={classNames(className, {
        'margin-top-6': newTab || newTabOnly
      })}
    >
      {newTab || newTabOnly ? (
        <>
          <Icon.Close className="margin-right-05 text-tbottom" />
          {t('close')}
        </>
      ) : (
        <Breadcrumbs
          items={[BreadcrumbItemOptions.HELP_CENTER]}
          customItem={text}
        />
      )}
    </Button>
  );
}
