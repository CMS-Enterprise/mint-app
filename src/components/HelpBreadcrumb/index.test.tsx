import React from 'react';
import { render } from '@testing-library/react';

import HelpBreadcrumb from './index';

describe('HelpBreadcrumb', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<HelpBreadcrumb type="Back" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
