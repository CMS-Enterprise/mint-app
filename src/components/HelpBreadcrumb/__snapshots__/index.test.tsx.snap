// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelpBreadcrumb > matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="usa-button usa-button--unstyled margin-top-6"
    data-testid="button"
    type="button"
  >
    <svg
      class="usa-icon margin-right-05 text-tbottom"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
    Close tab
  </button>
</DocumentFragment>
`;
